packages/slice-machine/src/features/builder/AddStaticFieldButton.tsx

```diff
@@ -0,0 +1,92 @@
+import { Box } from "@prismicio/editor-ui";
+import { createContext, ReactNode, useContext, useState } from "react";
+
+import {
+  AddFieldDropdown,
+  type AddFieldDropdownProps,
+} from "@/features/builder/AddFieldDropdown";
+import { StaticFieldsInfoDialog } from "@/features/builder/StaticFieldsInfoDialog";
+import { usePersistedState } from "@/hooks/usePersistedState";
+
+const LOCAL_STORAGE_KEY = "staticFieldsInfoDialogDismissed";
+
+export function AddStaticFieldButton(props: AddFieldDropdownProps) {
+  return (
+    <StaticFieldsInfoProvider>
+      <AddStaticFieldButtonComponent {...props} />
+    </StaticFieldsInfoProvider>
+  );
+}
+
+interface StaticFieldsInfoProviderProps {
+  children: ReactNode;
+}
+
+function StaticFieldsInfoProvider(props: StaticFieldsInfoProviderProps) {
+  const { children } = props;
+
+  const [isInfoDialogSeen, setInfoDialogSeen] = usePersistedState<boolean>(
+    LOCAL_STORAGE_KEY,
+    false,
+  );
+
+  return (
+    <StaticFieldsInfoContext.Provider
+      value={{ isInfoDialogSeen, setInfoDialogSeen }}
+    >
+      {children}
+    </StaticFieldsInfoContext.Provider>
+  );
+}
```

Comment:
💡 

I don't really see a reason to use context here, do we plan on open the modal from somewhere else? Seems like having the `usePersistedState` be use directly in the `AddStaticFieldButtonComponent` would suffice.

Reply:
I had some issue with that, but will look into it again!

Reply:
removed!